woocommerce/woocommerce-gateway-stripe
# Adds "Disable Stripe" confirmation modal

# Changes proposed in this Pull Request:

Fixes #1681

This PR adds a "disable Stripe" confirmation modal when a user attempts to disable the Stripe payment method when UPE is enabled and they have 1 or more payment methods added or express checkouts are enabled. If the user cancels the modal, the gateway remains enabled, if they confirm it the gateway is disabled.

# Testing instructions

From `/wp-admin/admin.php?page=wc-settings&tab=checkout`:
1. With UPE **disabled**, click the toggle to disable the `Stripe` method - no modal should appear.

From `/wp-admin/admin.php?page=wc-settings&tab=checkout`:
1. With UPE **enabled**, click `Stripe UPE` to manage settings.
2. Remove all payment methods.
3. Disable `Express Checkouts`.
4. Go back to `/wp-admin/admin.php?page=wc-settings&tab=checkout`
5. Click the toggle to disable the `Stripe` method - no modal should appear.

From `/wp-admin/admin.php?page=wc-settings&tab=checkout`:
1. With UPE **enabled**, click `Stripe UPE` to manage settings.
2. Remove all payment methods.
3. Enable `Express Checkouts`.
4. Go back to `/wp-admin/admin.php?page=wc-settings&tab=checkout`
5. Click the toggle to disable the `Stripe` method - the confirmation modal should appear.

From `/wp-admin/admin.php?page=wc-settings&tab=checkout`:
1. With UPE **enabled**, click `Stripe UPE` to manage settings.
2. Add at least 1 payment method.
4. Go back to `/wp-admin/admin.php?page=wc-settings&tab=checkout`
5. Click the toggle to disable the `Stripe` method - the confirmation modal should appear.
